packages/rspack/src/config/zod/target.ts

```diff
@@ -1,21 +1,59 @@
 import { z } from "zod";
 
-const allowTarget = z.enum([
-	"node",
```

Comment:
I would suggest using string literal, which would infer a more accurate `type`. Users could use the type for auto-completion.

Reply:
I'm ok with using `z.custom` on the `electron` part. I think it's better to list all the possible `es` targets, so people wouldn't write something like `es6`

Reply:
you can write `es6` in webpack 🤷

Reply:
<img width="355" alt="Screenshot 2023-06-05 at 14 29 42" src="https://github.com/web-infra-dev/rspack/assets/42857895/1fc05a14-8a86-42dd-944a-8e6da6e1703b">
ok this is bad